src/store/atoms/markAsComplete.ts

```diff
@@ -0,0 +1,12 @@
+import {atom} from 'recoil';
+
+interface markAsCompleteParams {
+    isValid : boolean
+    path? : string,
+    isCompleted? : boolean
+}
+
+export const markAsCompleteAtom = atom<markAsCompleteParams>({
```

Comment:
This approach seems slightly ad-hoc-ish
You're using a single atom for all checkboxes
You probably need an atom family here